### PR TITLE
attempt to fix baking issue

### PR DESCRIPTION
**Pull request by s-leger**
Fix for issue #588
keyframe_insert may return ValueError: bpy_struct.keyframe_insert() path spans ID blocks

### CODE REVIEW

File: nodes/object/object_data_path_output.py [hunk 1]
```diff
@@ -32,12 +32,24 @@ def execute(self, object, path, arrayIndex, value):
...
             self.errorMessage = "Error"
         return object
-
+    
+    def getPropertyPath(self, object, path):
+        if "." in path:
+            propPathStr, propName = path.rsplit(".", 1)
+            propPath = object.path_resolve(propPathStr)
+        else:
+            propPath = object
```
Comment (JacquesLucke) [inline nodes/object/object_data_path_output.py]: the name "propPath" is very misleading as it contains an object

**Comment by s-leger** [comment]
Well, maybe dataPath could be better, as with material the path resolve in  bpy.data.materials ?

File: nodes/object/object_data_path_output.py [hunk 2]
```diff
@@ -32,12 +32,24 @@ def execute(self, object, path, arrayIndex, value):
         except:
             self.errorMessage = "Error"
         return object
-
+    
+    def getPropertyPath(self, object, path):
+        if "." in path:
+            propPath, propName = path.rsplit(".", 1)
+            dataPath = object.path_resolve(propPath)
```
Comment (JacquesLucke) [inline nodes/object/object_data_path_output.py]: can this throw an exception if the path is not valid?

File: nodes/object/object_data_path_output.py [hunk 3]
```diff
@@ -36,7 +36,10 @@ def execute(self, object, path, arrayIndex, value):
     def getPropertyPath(self, object, path):
         if "." in path:
             propPath, propName = path.rsplit(".", 1)
-            dataPath = object.path_resolve(propPath)
+            try:
```
Comment (s-leger) [inline nodes/object/object_data_path_output.py]: Prevent ValueError from rising if path is not valid

**Comment by JacquesLucke** [comment]
Looks good. Can you maybe give 2-3 examples of data paths that work now but did not work before?

**Comment by s-leger** [comment]
At least anything related to standard materials and textures, like data.materials[0].alpha and so on.
Not found any other one until now.

**Comment by JacquesLucke** [comment]
ok, so at least this works now :)
All things that worked before shouldn't be broken by this change. good job :)

**Pull request merged**